### PR TITLE
Update CI test matrix: Add Django 3.2, drop Python 3.6, Django 3.0

### PR DESCRIPTION
**Pull request by amureki**
In 3.2 primary_keys behaviour changed a bit:
https://docs.djangoproject.com/en/3.2/releases/3.2/#customizing-type-of-auto-created-primary-keys

This is affecting our tests, need to dig deeper into it.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
 
 [testenv]
 deps=
     -rrequirements-dev.txt
-    dj22: django>=2.2a1,<3.0
-    dj30: django>=3.0a1,<3.1
-    dj31: django>=3.1a1,<3.2
+    dj22: Django==2.2
+    dj31: Django>=3.1,<3.2
+    dj32: Django>=3.2,<3.3
```
Comment (codingjoe) [inline tox.ini:19]: You could use the new `~=` version syntax if you want the latest bugfix release of a minor version.

**Comment by amureki** [comment]
I created a bug-ticket, let's see if maintainers will acknowledge that:
https://code.djangoproject.com/ticket/32733#ticket

**Comment by amureki** [comment]
@syphar while playing around with backend yesterday, I think I noticed some glitch with Django 3.2 and one of our HC models.
I'll try to make a more thoughtful test today to see if it was a real issue or something minor.

**Comment by syphar** [comment]
@amureki can we work around this issue?

**Comment by amureki** [comment]
@syphar sadly I did not test it yet. So, what we need to validate:
- [ ] we can deploy backend with Django 3.2 and django-heroku-connect 0.24.0 (deployments checks are passing and mapping reimported)
- [ ] Heroku Connect models are working fine (updating/saving)

We can do this on UAT. If everything looks fine, then we can disable/ignore failing tests for Django 3.2 until there would be a resolution on https://code.djangoproject.com/ticket/32733#ticket

**Comment by amureki** [comment]
https://code.djangoproject.com/ticket/32733#ticket my patch was accepted and merged to main. It would be released in Django 3.2.4 (so far, it is planned on June 1st, but the date could be moved).

**Comment by syphar** [comment]
just ran the tests again, in one place I had to remove the `abstract=True` from a test-model because it is not allowed to create instances of abstract models any more. 

Then, a next test failure, where I'm not sure if this another bug in Django, or something we have to fix. 

```
___________________ TestHerokuConnectModelMixin.test_check_sf_object_name ___________________
tests/db/models/test_base.py:251: in test_check_sf_object_name
    errors = MyModel.check()
heroku_connect/db/models/base.py:303: in check
    errors = super().check(**kwargs)
.direnv/python-3.9.4/lib/python3.9/site-packages/django/db/models/base.py:1293: in check
    *cls._check_default_pk(),
.direnv/python-3.9.4/lib/python3.9/site-packages/django/db/models/base.py:1309: in _check_default_pk
    not cls._meta.app_config._is_default_auto_field_overridden
E   AttributeError: 'NoneType' object has no attribute '_is_default_auto_field_overridden'
```

before I invest time to dig deeper, can anyone already answer the question?

**Comment by syphar** [comment]
Never mind, I worked around it.

**Comment by syphar** [comment]
@amirnaeimi for you to review, please :) 

( also I'm happy to implement comments by @amureki or @codingjoe , if they have any :) )

**Review by amirnaeimi — APPROVED** [review]

**Pull request merged**